zebra/zebra_nhg.h

```diff
@@ -24,6 +24,245 @@
 #define __ZEBRA_NHG_H__
 
 #include "zebra/rib.h"
+#include "lib/nexthop_group.h"
+
+#include "zebra/zebra_dplane.h"
+
+/* This struct is used exclusively for dataplane
+ * interaction via a dataplane context.
+ *
+ * It is designed to mimic the netlink nexthop_grp
+ * struct in include/linux/nexthop.h
+ */
+struct nh_grp {
```

Comment:
why? why not add ... a weight to the main group struct and do the conversion from netlink to os-neutral?

Reply:
I am not following, this is an OS neutral one right? I needed this to be able to store a information about a group we received from the kernel and enque it onto the rib metaqueue for processing later.

I did this to account for the future dataplane enhancements so all routes from the kernel are received in the other ptrhead as well. Rather than creating nhg_hash_entry objects in the netlink code and adding them directly from there.

I don't think this belongs as part of the main group struct since its really only for queuing a group to be handled later.